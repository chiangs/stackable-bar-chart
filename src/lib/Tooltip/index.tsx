import * as React from "react";
import { BarData } from "../__types";

type Props = BarData;

const NAME_COMPONENT = "tooltip-container";

const Tooltip: React.FC<Props> = ({
  label,
  value,
  color,
  background,
  percentage,
}) => {
  const style = { background, color };
  return (
    <div className={NAME_COMPONENT} style={style}>
      <span>
        {value} {label} - {`${percentage}%`}
      </span>
    </div>
  );
};

export default Tooltip;
