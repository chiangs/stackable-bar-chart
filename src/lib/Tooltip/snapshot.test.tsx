import * as React from "react";
import renderer from "react-test-renderer";
import { describe, expect, test } from "vitest";
import type { BarData } from "../__types";
import Tooltip from ".";

describe("Tooltip", () => {
  test("Tooltip component renders correctly", () => {
    const props: BarData = {
      label: "test",
      value: 1,
    };
    const component = renderer.create(<Tooltip {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
