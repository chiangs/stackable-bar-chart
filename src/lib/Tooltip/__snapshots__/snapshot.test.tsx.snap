// Vitest Snapshot v1

exports[`Tooltip > Tooltip component renders correctly 1`] = `
<div
  className="tooltip-container"
  style={
    {
      "background": undefined,
      "color": undefined,
    }
  }
>
  <span>
    1
     
    test
     - 
    undefined%
  </span>
</div>
`;
